wdl-engine/src/stdlib/read_tsv.rs

```diff
@@ -0,0 +1,480 @@
+use wdl_ast::Diagnostic;
+use wdl_grammar::lexer::v1::is_ident;
+
+use super::CallContext;
+use super::Function;
+use super::Signature;
+use crate::Array;
+use crate::CompoundValue;
+use crate::PrimitiveValue;
+use crate::Value;
+use crate::diagnostics::function_call_failed;
+
+/// Represents a header in a TSV (tab-separated value) file.
+enum TsvHeader {
+    /// The header was explicitly specified as an `Array[String]`.
+    Specified(Array),
+    /// The header was read from the file.
+    File(String),
+}
+
+impl TsvHeader {
+    /// Gets the column names in the header.
+    ///
+    /// # Panics
+    ///
+    /// Panics if a specified header contains a value that is not a string.
+    pub fn columns(&self) -> impl Iterator<Item = &str> {
+        match self {
+            Self::Specified(array) => Either::Left(array.elements().iter().map(|v| {
+                v.as_string()
+                    .expect("header value must be a string")
+                    .as_str()
+            })),
+            Self::File(s) => Either::Right(s.split('\t')),
+        }
+    }
+}
+
+/// Reads a tab-separated value (TSV) file as an Array[Array[String]]
+/// representing a table of values.
+///
+/// Trailing end-of-line characters (\r and \n) are removed from each line.
+///
+/// `Array[Array[String]] read_tsv(File, [false])``: Returns each row of the
+/// table as an Array[String]. There is no requirement that the rows of the
+/// table are all the same length.
+///
+/// https://github.com/openwdl/wdl/blob/wdl-1.2/SPEC.md#read_tsv
+fn read_tsv_simple(context: CallContext<'_>) -> Result<Value, Diagnostic> {
+    debug_assert!(context.arguments.len() == 1);
+    debug_assert!(context.return_type_eq(ANALYSIS_STDLIB.array_array_string_type()));
+
+    let path = context.cwd().join(
+        context
+            .coerce_argument(0, PrimitiveTypeKind::File)
+            .unwrap_file()
+            .as_str(),
+    );
+
+    let file = fs::File::open(&path)
+        .with_context(|| format!("failed to open file `{path}`", path = path.display()))
+        .map_err(|e| function_call_failed("read_tsv", format!("{e:?}"), context.call_site))?;
+
+    let mut rows: Vec<Value> = Vec::new();
+    for line in BufReader::new(file).lines() {
+        let line = line
+            .with_context(|| format!("failed to read file `{path}`", path = path.display()))
+            .map_err(|e| function_call_failed("read_tsv", format!("{e:?}"), context.call_site))?;
+        let values = line
+            .trim_end_matches(['\r', '\n'])
+            .split('\t')
+            .map(|s| PrimitiveValue::new_string(s).into())
+            .collect::<Vec<Value>>();
+        rows.push(
+            Array::new_unchecked(ANALYSIS_STDLIB.array_string_type(), Arc::new(values)).into(),
+        );
+    }
+
+    Ok(Array::new_unchecked(ANALYSIS_STDLIB.array_array_string_type(), Arc::new(rows)).into())
+}
+
+/// Reads a tab-separated value (TSV) file as an Array[Array[String]]
```

Comment:
Shouldn't this be:
```suggestion
/// Reads a tab-separated value (TSV) file as an Array[Object]
```
?